aws-iot-topicrule/resource-role.yaml

```diff
@@ -0,0 +1,39 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: >
+  This CloudFormation template creates a role assumed by CloudFormation
+  during CRUDL operations to mutate resources on behalf of the customer.
+
+Resources:
+  ExecutionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      MaxSessionDuration: 8400
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: resources.cloudformation.amazonaws.com
+            Action: sts:AssumeRole
+      Path: "/"
+      Policies:
+        - PolicyName: ResourceTypePolicy
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                - "iam:PassRole"
+                - "iot:CreateTopicRule"
+                - "iot:DeleteTopicRule"
+                - "iot:GetTopicRule"
+                - "iot:ListTagsForResource"
+                - "iot:ListTopicRules"
+                - "iot:ReplaceTopicRule"
+                - "iot:TagResource"
+                - "iot:UntagResource"
+                Resource: "*"
```

Comment:
This might be too permissive. We need to ask the CFN people if they are ok with this.